compiler/src/main/scala/org/scalajs/nscplugin/PrepJSExports.scala

```diff
@@ -441,55 +441,40 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
     }
   }
 
-  /** generate an exporter for a DefDef including default parameter methods */
-  private def genExportDefs(defSym: Symbol, jsName: String, pos: Position) = {
-    val clsSym = defSym.owner
-    val scalaName =
-      jsInterop.scalaExportName(jsName, jsInterop.isJSProperty(defSym))
+  /** generate an exporter for a target including default parameter methods */
+  private def genExportDefs(sym: Symbol, jsName: String, pos: Position) = {
+    val clsSym =
+      if (sym.isConstructor) sym.owner.owner
+      else sym.owner
+
+    val isProperty = sym.isModuleClass || isJSAny(sym) || jsInterop.isJSProperty(sym)
+    val scalaName = jsInterop.scalaExportName(jsName, isProperty)
 
     // Create symbol for new method
-    val expSym = defSym.cloneSymbol
-
-    // Set position of symbol
-    expSym.pos = pos
-
-    // Alter type for new method (lift return type to Any)
-    // The return type is lifted, in order to avoid bridge
-    // construction and to detect methods whose signature only differs
-    // in the return type.
-    // Attention: This will cause boxes for primitive value types and value
-    // classes. However, since we have restricted the return types, we can
-    // always safely remove these boxes again in the back-end.
-    if (!defSym.isConstructor)
-      expSym.setInfo(retToAny(expSym.tpe))
-
-    // Change name for new method
-    expSym.name = scalaName
-
-    // Update flags
-    expSym.setFlag(Flags.SYNTHETIC)
-    expSym.resetFlag(
-        Flags.DEFERRED     | // We always have a body
-        Flags.ACCESSOR     | // We are never a "direct" accessor
-        Flags.CASEACCESSOR | // And a fortiori not a case accessor
-        Flags.LAZY         | // We are not a lazy val (even if we export one)
-        Flags.OVERRIDE       // Synthetic methods need not bother with this
-    )
-
-    // Remove export annotations
-    expSym.removeAnnotation(JSExportAnnotation)
-
-    // Add symbol to class
-    clsSym.info.decls.enter(expSym)
+    val expSym = clsSym.newMethod(scalaName, pos, Flags.SYNTHETIC)
+    val expSymTpe = {
+      /* Alter type for new method (lift return type to Any)
+       * The return type is lifted, in order to avoid bridge
+       * construction and to detect methods whose signature only differs
+       * in the return type.
+       * Attention: This will cause boxes for primitive value types and value
+       * classes. However, since we have restricted the return types, we can
+       * always safely remove these boxes again in the back-end.
```

Comment:
Is this part of the comment still relevant today? I don't understand what it refers to. 🤔 

It appears to be 9 years old, dating all the way to the original implementation of `@JSExport` in aa4df92a6d08ba0f06b6d8f5075427a5c1ffeee8. At that time, apparently we restricted result types:
https://github.com/scala-js/scala-js/commit/aa4df92a6d08ba0f06b6d8f5075427a5c1ffeee8#diff-21a703ddaf33256766f84be3252c7207fe2c9b7616ff3da9a5b70c7780b666fdR121

Reply:
I think you are right. This just needs to be removed. Especially since we *want* boxes for primitive and value types here (mostly value types, but still...).